lib/puma/server.rb

```diff
@@ -422,11 +420,12 @@ def handle_servers
 
         @events.fire :state, @status
 
-        graceful_shutdown if @status == :stop || @status == :restart
         if queue_requests
```

Comment:
I presume this needs to be @queue_requests given your change. But I'd also rather you not change this to an ivar read and leave it as a local var.

Reply:
There were actually two separate `queue_requests` local vars- one within `Server#run`, and one within `Server#handle_servers`.

The local var in `#run` had to be removed in order for this PR to work at all- see my comment at https://github.com/puma/puma/pull/2122#pullrequestreview-362317379. However, this one in `#handle_servers` actually doesn't need to be changed as well, so I left it as-is to minimize the diff.

That said, since this local var in `#handle_servers` is only ever read on server shutdown and probably has no performance impact, I would lean towards removing it for consistency/simplicity if that's your preference.

Reply:
This explanation works for me.